src/sentry/integrations/slack/notify_action.py

```diff
@@ -182,13 +182,21 @@ def get_channel_id(self, integration_id, name):
             'exclude_members': True,
         })
 
-        resp = session.get('https://slack.com/api/channels.list', params=channels_payload)
-        resp = resp.json()
-        if not resp.get('ok'):
-            self.logger.info('rule.slack.channel_list_failed', extra={'error': resp.get('error')})
-            return None
+        # Slack limits the response of `channels.list` to 1000 channels, paginate if needed
+        channels = []
+        cursor = None
+        while cursor or len(channels) == 0:
+            resp = session.get('https://slack.com/api/channels.list', params=dict(channels_payload, **{
+                'cursor': cursor
+            }))
+            resp = resp.json()
+            if not resp.get('ok'):
+                self.logger.info('rule.slack.channel_list_failed', extra={'error': resp.get('error')})
+                return None
+            channels.extend(resp['channels'])
+            cursor = resp.get('response_metadata', {}).get('next_cursor', None)
 
-        channel_id = {c['name']: c['id'] for c in resp['channels']}.get(name)
+        channel_id = {c['name']: c['id'] for c in channels}.get(name)
```

Comment:
Hi @cjmaio! Sorry for the delay on this. Yes, we should paginate.

Can we change it to check if the reuqested channel name exists after each page, that way we don't needlessly continue paging when we could have found the ID in the first page (for example).